### PR TITLE
Interpretation regression tests and benchmarks

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

- Adds interpreter regression testing with a scan backup
- Caches generated ballot fixtures to speed up interpretation tests
- Adds some basic interpretation benchmarks (which are _not_ integrated into CI due to flakiness)

_Review by commits_

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

**Comment by jonahkagan** [comment]
@arsalansufi I'd be curious to see how the benchmarks compare on your machine

File: libs/ballot-interpreter/src/interpret_bmd_ballots.test.ts
Comment (jonahkagan) [inline libs/ballot-interpreter/src/interpret_bmd_ballots.test.ts:1]: No code changes, just copied out of interpret.test.ts

File: libs/ballot-interpreter/src/interpret_all_bubble_ballot.test.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  test('Blank ballot interpretation', async () => {
+    const ballotImagePaths = await ballotPdfToPageImages(blankBallotPath);
+    const [frontResult, backResult] = await interpretSheet(
+      {
+        electionDefinition,
+        precinctSelection: singlePrecinctSelectionFor(precinctId),
+        testMode: true,
+      },
+      ballotImagePaths
+    );
```
Comment (jonahkagan) [inline libs/ballot-interpreter/src/interpret_all_bubble_ballot.test.ts:35]: This is basically the pattern of change that I made throughout this commit. The assertions are left unchanged

File: libs/ballot-interpreter/src/interpret_nh_hmpb.test.ts
Comment (jonahkagan) [inline libs/ballot-interpreter/src/interpret_nh_hmpb.test.ts:1]: No code changes, just copied out of interpret.test.ts

File: libs/ballot-interpreter/benchmarks/benchmarking.ts [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    iter(measurements)
+      .map((x) => (x - mean) ** 2)
+      .sum() /
+    (measurements.length - 1);
+  const standardDeviation = Math.sqrt(variance);
+  const median = assertDefined(
+    [...measurements].sort()[Math.floor(measurements.length / 2)]
+  );
+  // 99% confidence interval
+  // See https://www.mathsisfun.com/data/confidence-interval.html
```
Comment (arsalansufi) [inline libs/ballot-interpreter/benchmarks/benchmarking.ts:49]: `mathsisfun.com` 😆

**Review by arsalansufi — APPROVED** [review]
LGTM!

File: libs/ballot-interpreter/README.md [hunk 3]
```diff
@@ -47,6 +47,44 @@ bin/interpret path/to/workspace d34d-b33f
...
+To run benchmarks for interpretation, run:
+
+```sh
+pnpm benchmark
+```
+
+This will run a series of benchmark tests, which will interpret a ballot many
+times and compute some statistics about the time it takes. The results will be
+compared to saved results from a previous run, and, if they differ by too large
+an amount, show a test failure message.
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply libs/ballot-interpreter/README.md:67]: Ah, well that's a bit of a bummer. I guess for now we'll have to have personal benchmarks then 😆

File: libs/ballot-interpreter/benchmarks/benchmarking.ts (hunk 2 above)
Comment (jonahkagan) [reply libs/ballot-interpreter/benchmarks/benchmarking.ts:49]: Honestly had the most approachable explanations of any resource I found lol

**Pull request merged**